### PR TITLE
Add terradoc

### PR DESCRIPTION
**Pull request by kassianh**
Recreated README with terradoc, checked documentation for errors or inconsistencies.

### CODE REVIEW

**Comment by kassianh** [comment]
@mariux I'm not too sure why the test is failing. Would you be able to have a look? Thank you.

File: README.tfdoc.hcl [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+              Sets the order in which rules are evaluated, lowest to highest.
+                  A lifecycle policy rule with a priority of `1` will be acted upon first,
+                  a rule with priority of `2` will be next, and so on.
+                  When you add rules to a lifecycle policy,
+                  you must give them each a unique value for `rulePriority`.
+                  Values do not need to be sequential across rules in a policy.
+                  A rule with a `tagStatus` value of any must have the highest value for `rulePriority` and be evaluated last.
+                  **Note:** The AWS ECR API seems to reorder rules based on `rulePriority`.
+                  If you define multiple rules that are not sorted in ascending `rulePriority` order in the Terraform code,
+                  the resource will be flagged for recreation every terraform plan.
```
Comment (mariux) [inline README.tfdoc.hcl:316]: wrong indent

File: README.md [hunk 2]
```diff
@@ -15,20 +15,20 @@ A [Terraform] base module for creating an
...
   - [Top-level Arguments](#top-level-arguments)
+    - [Module Configuration](#module-configuration)
     - [Main Resource Configuration](#main-resource-configuration)
     - [Extended Resource configuration](#extended-resource-configuration)
     - [Additional configuration](#additional-configuration)
-    - [`policy_statements` Object Arguments](#policy_statements-object-arguments)
-    - [`principal` Object Arguments](#principal-object-arguments)
-    - [`lifecycle_rules` Object Arguments](#lifecycle_rules-object-arguments)
-    - [`selection` Object Arguments](#selection-object-arguments)
-    - [`action` Object Arguments](#action-object-arguments)
```
Comment (soerenmartius) [inline README.md:29]: why is this getting removed? is this expected?

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+          description = <<-END
+            Specifies whether resources in the module will be created.
+          END
+        }
+
+        variable "module_tags" {
+          type        = map(string)
+          default     = {}
+          description = <<-END
+            A map of tags that will be applied to all created resources that accept tags. Tags defined with 'module_tags' can be overwritten by resource-specific tags.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:170]:
```suggestion
            A map of tags that will be applied to all created resources that accept tags. Tags defined with `module_tags` can be overwritten by resource-specific tags.
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (kassianh) [reply README.md:29]: @soerenmartius Yes it is. The arguments have been moved from their own extra sections like in the above table of contents, to the variables themselves. It's all still there, just no longer as an extra section.

**Review by soerenmartius — APPROVED** [review]
lgtm

**Pull request merged**